### PR TITLE
WRP-6598: Add vertical layout

### PR DESCRIPTION
**Pull request by alexandrumorariu**
### Checklist

* [X] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [X] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
I added an orientation option and made some layout changes to facilitate a vertical transfer list component.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRP-6598

### CODE REVIEW

File: TransferList/TransferList.js [hunk 1]
```diff
@@ -273,13 +314,31 @@ const TransferListBase = kind({
...
+			const handleSpotlightRight = (ev) => {
+				if (orientation === 'horizontal' && list === 'first') moveInSecond();
+				if ((elements.length - 1 !== index && elements.length - 2 !== index) || orientation === 'horizontal') return;
 				ev.preventDefault();
 				ev.stopPropagation();
 			};
 
 			const handleSpotlightUp = (ev) => {
-				if (index !== 0) return;
+				if (orientation !== 'horizontal' && list === 'second') moveInFirst();
```
[the remark this answers is not in the document]
Comment (alexandrumorariu) [reply TransferList/TransferList.js]: Done

File: TransferList/TransferList.js [hunk 2]
```diff
@@ -730,74 +792,112 @@ const TransferListBase = kind({
...
 			onSelect: setSelected,
 			selectedItems,
-			onSpotlightLeft: handleSpotlightLeft,
+			moveInFirst,
 			showSelectionOrder
 		};
 
 		return (
-			<Layout align="center" className={componentCss.transferList}>
+			<Layout align="center" className={componentCss.transferList} orientation={orientation !== 'horizontal' ? 'vertical' : 'horizontal'}>
```
[the remark this answers is not in the document]
Comment (alexandrumorariu) [reply TransferList/TransferList.js]: Done

**Review by daniel-stoian-lgp — APPROVED** [review]

File: TransferList/TransferList.js [hunk 3]
```diff
@@ -140,6 +140,15 @@ const TransferListBase = kind({
 		 */
 		noMultipleDrag: PropTypes.bool,
 
+		/**
+		 * The orientation for the transfer list.
+		 *
+		 * @type {Component}
```
Comment (adrian-cocoara-lgp) [inline TransferList/TransferList.js:146]:
```suggestion
		 * @type {('horizontal', 'vertical')}
```

see Button example:
image

**Comment by adrian-cocoara-lgp** [comment]
Only one comment for jsdoc. Good job overall! 👍

**Review by adrian-cocoara-lgp — APPROVED** [review]
LGTM

**Pull request merged**